File: samples/echo-java-enclave/src/main/java/com/r3/sgx/enclave/jecho/EchoHandler.java
package com.r3.sgx.enclave.jecho;

import com.r3.sgx.core.common.Handler;
import com.r3.sgx.core.common.Sender;

import java.nio.ByteBuffer;

class EchoHandler implements Handler<Sender> {
    @Override
    public Sender connect(Sender upstream) {
        return upstream;
    }

    @Override
    public void onReceive(Sender sender, ByteBuffer input) {
        sender.send(input.remaining(), buffer -> buffer.put(input));
    }
}

File: enclave-testing/src/main/kotlin/com/r3/sgx/testing/EchoHandler.kt
package com.r3.sgx.testing

import com.r3.sgx.core.common.Handler
import com.r3.sgx.core.common.Sender
import java.nio.ByteBuffer
import java.util.function.Consumer

class EchoHandler : Handler<Sender> {
    override fun onReceive(connection: Sender, input: ByteBuffer) {
        connection.send(input.remaining(), Consumer { buffer ->
            buffer.put(input)
        })
    }

    override fun connect(upstream: Sender) = upstream
}